lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,285 @@
+          extend(InsertImplementation)
+        when :load
+          raise Fluent::ConfigError, "'template_suffix' is for only `insert` mode, instead use 'fetch_schema_table' and formatted table name" if @template_suffix
+          extend(LoadImplementation)
         end
-      when :application_default
-        # Do nothing
-      else
-        raise Fluent::ConfigError, "unrecognized 'auth_method': #{@auth_method}"
-      end
-
-      unless @table.nil? ^ @tables.nil?
-        raise Fluent::ConfigError, "'table' or 'tables' must be specified, and both are invalid"
-      end
 
-      @tablelist = @tables ? @tables.split(',') : [@table]
-
-      @fields = Fluent::BigQuery::RecordSchema.new('record')
-      if @schema_path
-        @fields.load_schema(MultiJson.load(File.read(@schema_path)))
-      end
+        case @auth_method
+        when :private_key
+          unless @email && @private_key_path
+            raise Fluent::ConfigError, "'email' and 'private_key_path' must be specified if auth_method == 'private_key'"
+          end
+        when :compute_engine
+          # Do nothing
+        when :json_key
+          unless @json_key
+            raise Fluent::ConfigError, "'json_key' must be specified if auth_method == 'json_key'"
+          end
+        when :application_default
+          # Do nothing
+        else
+          raise Fluent::ConfigError, "unrecognized 'auth_method': #{@auth_method}"
+        end
 
-      types = %w(string integer float boolean timestamp)
-      types.each do |type|
-        raw_fields = instance_variable_get("@field_#{type}")
-        next unless raw_fields
-        raw_fields.split(',').each do |field|
-          @fields.register_field field.strip, type.to_sym
+        unless @table.nil? ^ @tables.nil?
+          raise Fluent::ConfigError, "'table' or 'tables' must be specified, and both are invalid"
         end
-      end
 
-      @regexps = {}
-      (1..REGEXP_MAX_NUM).each do |i|
-        next unless conf["replace_record_key_regexp#{i}"]
-        regexp, replacement = conf["replace_record_key_regexp#{i}"].split(/ /, 2)
-        raise ConfigError, "replace_record_key_regexp#{i} does not contain 2 parameters" unless replacement
-        raise ConfigError, "replace_record_key_regexp#{i} contains a duplicated key, #{regexp}" if @regexps[regexp]
-        @regexps[regexp] = replacement
-      end
+        @tablelist = @tables ? @tables : [@table]
 
-      @localtime = false if @localtime.nil? && @utc
+        @fields = Fluent::BigQuery::RecordSchema.new('record')
+        if @schema_path
+          @fields.load_schema(MultiJson.load(File.read(@schema_path)))
+        end
 
-      @timef = TimeFormatter.new(@time_format, @localtime)
+        types = %w(string integer float boolean timestamp)
+        types.each do |type|
+          fields = instance_variable_get("@field_#{type}")
+          next unless fields
+          fields.each do |field|
+            @fields.register_field field.strip, type.to_sym
```

Comment:
On the parse of `array` type, whitespaces around comma will be stripped if the value is simple String.
And, if user wants to include whitespaces at the beginning/end of values, he can use json style like `[" value with space ", " yay "]`.
So this line should NOT do `field.strip`. It's unneeded for simple String style, nor worse for Json style.